examples/algorithms/bc/bc.cu

```diff
@@ -54,51 +53,63 @@ void test_bc(int num_arguments, char** argument_array) {
   // --
   // GPU Run
 
+  size_t n_runs = source_vect.size();
   std::vector<float> run_times;
-  for (int i = 0; i < source_vect.size(); i++) {
-    // Record run times without collecting metrics (due to overhead)
-    run_times.push_back(gunrock::bc::run(G, source_vect[i], true,
-                                         bc_values.data().get(), &edges_visited,
-                                         &search_depth));
-  }
-
-  // --
-  // Log
 
-  std::cout << "Single source : " << source_vect.back() << "\n";
-  print::head(bc_values, 40, "GPU bc values");
-  std::cout << "GPU Elapsed Time : " << run_times[params.num_runs - 1]
-            << " (ms)" << std::endl;
-
-  // --
-  // Run performance evaluation
+#if !(ESSENTIALS_COLLECT_METRICS)
```

Comment:
The main change I see is that we don't need this #if, #else, #endif. It can just be a part of the `benchmark::INIT_BENCH();`. When the variable is set, the init does something, otherwise it is just empty.

This way, we can always have `benchmark::INIT_BENCH();` there with no if/else branch and duplicated code.

Reply:
```cpp
void INIT_BENCH() {
#if ESSENTIALS_COLLECT_METRICS
  thrust::fill(____.edges_visited.begin(), ____.edges_visited.end(), 0);
  thrust::fill(____.vertices_visited.begin(), ____.vertices_visited.end(), 0);

  BXXX.d_edges_visited = ____.edges_visited.data().get();
  BXXX.d_vertices_visited = ____.vertices_visited.data().get();
#endif
}
```